### PR TITLE
Take context Done into considerations in a bunch of places

### PR DESCRIPTION
**Pull request by mstoykov**
This was leaving hanging goroutines all over the place. 

Not certain whether any of those are relevant for real running, but in tests it was both making stack traces harder to read (as you have a bunch of blocked goroutines), but it also likely was taking some CPU/MEM away from tests. The later likely isn't all that much, but still better to not have goroutines that hang around forever ;)

### CODE REVIEW

**Review by ankur22 — APPROVED** [review]
LGTM, but i'd wait for an approval from someone else first 🙂

**Review by imiric — APPROVED** [review]
LGTM, seems safe to do in general.

How did you spot this @mstoykov? Just from the stack traces?

**Review by inancgumus — APPROVED** [review]
Ran some tests, and it looks harmless.

> in tests it was both making stack traces harder to read (as you have a bunch of blocked goroutines

However, I couldn't see much difference between the stack traces of `main` and this fix. Out of curiosity, which goroutines you found were hanging out?

Test:

```go
func TestFramePress(t *testing.T) {
	ctx, cancel := context.WithCancel(context.Background()) // added
	tb := newTestBrowser(t, ctx) // added

	p := tb.NewPage(nil)

	p.SetContent(`<input id="text1">`, nil)

	f := p.Frames()[0]

	f.Press("#text1", "Shift+KeyA", nil)
	cancel() // added
	f.Press("#text1", "KeyB", nil)
	f.Press("#text1", "Shift+KeyC", nil)

	require.Equal(t, "AbC", f.InputValue("#text1", nil))
}
```

**Comment by mstoykov** [comment]
> How did you spot this @mstoykov? Just from the stack traces?

yeah - if you run with `-count 10 -race` for example - it usually turns out that something hangs and then in the stacktraces I would have like a page or 20 of just goroutines that are stuck in the places changed here.

This is also is how you can get ... more or less always some kind of failure of the tests.

> However, I couldn't see much difference between the stack traces of main and this fix. Out of curiosity, which goroutines you found were hanging out?

The ones that are running the particular code that is changed - they just sometimes would not be able to send their message as the channel won't be listened to as the context was already done at this point.

You likely will need to run the test multiple times - preferably hitting some other issues in the meantime that dumps all the goroutines. And then what you get is a  lot of those goroutines that are just hanging around, but AFAIK don't actually block anything else. So given that they likely just need to stop as the context was done - I just added the `select`s and now the stack traces are a lot smaller.

**Pull request merged**
